jetty-server/src/main/java/org/eclipse/jetty/server/session/Session.java

```diff
@@ -1019,6 +1013,18 @@ protected SessionData getSessionData()
     public void setResident (boolean resident)
     {
         _resident = resident;
+        
+        if (_resident)
+            updateInactivityTimer();
```

Comment:
if _timer == null then this has been called with false before, so we should throw an ISE